src/Propulsion.Cosmos/PropulsionCosmosPrometheus.fs

```diff
@@ -4,111 +4,155 @@ namespace Propulsion.CosmosStore.Prometheus
 namespace Propulsion.Cosmos.Prometheus
 #endif
 
+(*
+    This file implements a Serilog Sink `LogSink`
+    which publishes metric values to Prometheus.
+
+    It takes in an additional set of custom tags
+    to annotate the metric we're publishing.
+
+*)
+
+type TagNames = string array
+type TagValues = string array
+type Tags = TagNames * TagValues
+
+[<AutoOpen>]
 module private Impl =
 
     let baseName stat = "propulsion_changefeed_" + stat
     let baseDesc desc = "Propulsion CosmosDB: ChangeFeed " + desc
     let groupLabels = [| "app"; "db"; "con"; "group" |]
     let rangeLabels = [| "app"; "db"; "con"; "group"; "rangeId" |]
+    let [<Literal>] secondsStat = "_seconds"
+    let [<Literal>] rusStat = "_rus"
+    let [<Literal>] latencyDesc = " latency"
+    let [<Literal>] chargeDesc = " charge"
+
+    let append = Array.append
 
 module private Gauge =
 
-    let private mk (cfg : Prometheus.GaugeConfiguration) name help =
-        let g = Prometheus.Metrics.CreateGauge(name, help, cfg)
-        fun app (db, con, group, range) v -> g.WithLabels(app, db, con, group, range).Set(v)
-    let private config = Prometheus.GaugeConfiguration(LabelNames = Impl.rangeLabels)
-    let create stat desc = mk config (Impl.baseName stat) (Impl.baseDesc desc)
+    let private make (config : Prometheus.GaugeConfiguration) name desc =
```

Comment:
Hm. Renames should be applied 'upstream' in Equinox too - its important that all these modules are consistent in their approach (though I do like the changes)

Ideally, this reformatting/renaming work would be a separate PR so we can eyeball the Equinox PRs and Propulsion PRs to make sure they're doing the exact same thing (esp as there is no test coverage on this module so we're reliant on careful eyeballing)

Reply:
Everything else was using `desc` and this alone was using `help`.
We'll apply this upstream to equinox as well - too much duplication going on.